js/load-image.js

```diff
@@ -159,10 +162,15 @@
 
   loadImage.onload = function (img, event, file, url, callback, options) {
     revokeHelper(url, options)
-    loadImage.transform(img, options, callback, file, {
-      originalWidth: img.naturalWidth || img.width,
-      originalHeight: img.naturalHeight || img.height
-    })
+
+    try {
+      loadImage.transform(img, options, callback, file, {
+        originalWidth: img.naturalWidth || img.width,
+        originalHeight: img.naturalHeight || img.height
+      })
+    } catch (error) {
+      callback(error)
+    }
```

Comment:
@blueimp This should now hanlde any uncaught exceptions in any transformations, if I understood the system correctly 🤔

Reply:
Yes, that's the right location for the `try...catch` wrapper. :)

Reply:
Great. Any tests you would like to be added?

Reply:
I think it's fine to include this with just the existing tests.

Although it would be good to test this specific scenario (huge canvas sizes) as well, the tradeoff would likely be largely increased memory requirements for running the tests.

Reply:
One other item I'm still thinking about though is that the Promise executor already has an implicit `try...catch` around it, so it's only necessary for the callback style.
I might add this improvement after the merge though, since it requires some restructuring.

Reply:
hmh, I used the Promise API, but `.catch()` hanlder wasn't called 🤔

Reply:
You're right, the Promise executor cannot catch errors that happen asynchronously.
So everything is fine as is. :)